app/jobs/send_messages_received_yesterday_job.rb

```diff
@@ -0,0 +1,30 @@
+class SendMessagesReceivedYesterdayJob < ApplicationJob
+  queue_as :low
+
+  def perform
+    publishers_with_messages_received_yesterday.each do |publisher|
+      next unless publisher.email?
+
+      message_count = count_messages_for_publisher(publisher)
+      Publishers::MessageNotificationMailer.messages_received(publisher: publisher, message_count: message_count).deliver
+    end
+  end
+
+  private
+
+  def publishers_with_messages_received_yesterday
+    Publisher.distinct
+             .joins(organisations: { vacancies: { job_applications: { conversations: :messages } } })
+             .where(messages: { type: "JobseekerMessage", read: false })
+             .where(messages: { created_at: Date.yesterday.all_day })
```

Comment:
🙋‍♂️ 
As far as I understand, this is still doing a sequential scan over all the `messages` table, which can become a problem once the `messages` table grows.

I don't have the full picture of how the messages are being queried elsewhere, but if I assume this is the only bit of code doing the "search messages filtering by...", adding this partial index for unread messages would greatly improve this query (and the `count_messages_for_publisher` one) performance: 

```suggestion
add_index :messages, [:type, :created_at], where: "read = false", name: "index_messages_unread_on_type_created_at"
```